hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -554,10 +556,6 @@ private Map<String, String> convertToMetadata(Database database) {
   }
 
   Database convertToDatabase(Namespace namespace, Map<String, String> meta) {
-    if (!isValidateNamespace(namespace)) {
```

Comment:
why wouldn't we want to check namespace validity here?

Reply:
My Bad. Thanks for catching this.

Reply:
if the API expects a `NoSuchNamespaceException` then we can't replace it with a IAE. The exception behavior here should be consistent with `loadNamespaceMetadata()`